src/main/java/com/shapesecurity/functional/data/HashTable.java

```diff
@@ -117,11 +117,36 @@ protected HashTable(@Nonnull Hasher<K> hasher, int length) {
         return HashTable.emptyUsingIdentity();
     }
 
+    @Nonnull
+    public static <K, V> HashTable<K, V> fromUsingEquality(@Nonnull ImmutableList<Pair<K, V>> list) {
+        return HashTable.<K, V>emptyUsingEquality().putAll(list);
+    }
+
+    @Nonnull
+    public static <K, V> HashTable<K, V> fromUsingIdentity(@Nonnull ImmutableList<Pair<K, V>> list) {
+        return HashTable.<K, V>emptyUsingIdentity().putAll(list);
+    }
+
+    @Nonnull
+    public static <K, V> HashTable<K, V> from(@Nonnull Hasher<K> hasher, @Nonnull ImmutableList<Pair<K, V>> list) {
+        return HashTable.<K, V>empty(hasher).putAll(list);
+    }
+
     @Nonnull
     public final HashTable<K, V> put(@Nonnull K key, @Nonnull V value) {
         return this.put(key, value, this.hasher.hash(key));
     }
 
+    @Nonnull
+    public final HashTable<K, V> put(@Nonnull Pair<K, V> pair) {
```

Comment:
Does this actually come up?

Reply:
It's a bit more concise when adding lists of pairs -- not too relevant given the also added `putAll`. I could see it being useful in some cases when a function returns a `Pair`.